### ISSUE TITLE
Display quoted message above the user message

### ISSUE THREAD

**Issue by Otiel** [RocketChat/feature-requests#688]
When quoting a message using the "Quote" or "Reply" action, I think it would be better if the quoted message appeared above the user message, not below as it is done currently:

![image](https://user-images.githubusercontent.com/3851902/36475651-a21a41ec-16fb-11e8-8bba-545f81348779.png)

This was already asked [here](https://github.com/RocketChat/Rocket.Chat/issues/164#issuecomment-216991364), I'm creating this issue to trace this more easily.

**Comment by jpenna** [comment]
Configurable might be harder to implement, and this feature was requested more than 1 year ago. 

Is there any reason why someone would want to see the replied message **after** the reply?

**Comment by mmickelson** [comment]
I agree with this request. Reading the quote after the reply is confusing. It's much more readable to see something like:
> Question or quoted message

Answer/reply

Configurable would be nice but I imagine the majority of people would would prefer to see the quote below the reply so I would say implement the hard-coded behavior and if it's requested to have it configurable, add another feature request specifically for that.

**Comment by ankar84** [comment]
I agree with this request.
It will be so nice to add the **optional choice** of displaying quotes in the Telegram and WhatsApp style (quoted text above, answer below)
and in the Rocket.Chat style (where the answer is located to above, and the quoted message is lower than the answer).

**Comment by zedar187** [comment]
Bump :)

**Comment by infused-kim** [comment]
From me too. Please implement.

**Comment by neomag** [comment]
Please implement

**Comment by prigal** [comment]
Indeed it's a weird pattern, it should at least be configurable imo.

Upvoted.

**Comment by svedrenne** [comment]
Please implement "Reply below" or "Quoted message above reply". Very Nice To Have

Upvoted (with "thumbs up")

**Comment by kyletmiller** [comment]
thanks for a great platform! adding support for this request.

**Comment by ankar84** [comment]
@rodrigok Hello, Rodrigo!
Please take a look at that feature request.
A lot of users asked for it more then 2 years now.

**Comment by jaykobi** [comment]
Like it is now I sometimes see that there is a quote only after I read the reply.
Much better to display the quote first.

**Comment by helgator** [comment]
Does anyone comprehend the logic behind "quote after reply" and can explain it? Maybe we are all just using RC the wrong way.

**Comment by rodrigok** [comment]
@helgator no, it's wrong, we are refactoring the messages and it will be fixed. It was made like that because we used the attachments component to display the content, and attachments were only available below the message.

**Comment by ankar84** [comment]
> we are refactoring the messages and it will be fixed

Really glad to hear that! Thank you for that work!

**Comment by tmkis2** [comment]
Just to bump it again: Any news on this? Any workaround to get this behavior changed before it get implemented?

**Comment by engelgabriel** [comment]
@ggazzo we should add this as part of the message refactoring into react.

**Comment by prayingmantis1979** [comment]
I've got many complains from users about current style of quoting :(

**Comment by Feuermurmel** [comment]
If this is touched, I'd also love to see the original message truncated to e.g. a maximum of two lines. Right now, the message flow is sometimes completely disrupted when long messages are quoted. Please have a look at how [Telegram Messenger](https://telegram.org) is doing this. IMHO they have the best UI for replies, I use it a lot and I never have trouble.

**Comment by MrVonkey** [comment]
Would it be possible to temporarily add an additional CSS class to the message component for quotes until the message component is refactored? This would allow admins to place quotes above the messages with custom CSS, e.g. with flex, and would allow excluding file attachments from the custom CSS. At the moment we are using the following custom CSS, which messes up the display position of the username (below file attachments and quotes):

```css
.message {
  display: flex;
  flex-direction: column;
}

.message > * {
 order: 1; 
}

.message > div:not([class]) {
  order: 0;
}
```

Not optimal, but still an improvement for readability in my opinion.

**Comment by skjardenCode** [comment]
Since beginning of April 2022, I started using RocketChat. I was very confused by quoted messages below the answer. I also vote to change this. The reading direction (flow of chat) is top to bottom, so should be the quote-answer direction.

**Comment by ankar84** [comment]
Great news - https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3883 introduces proper direction of quotes on mobile!
Kudos to mobile team!
Still waiting for Web client implementation...

**Comment by EdSantos754** [comment]
C'mon guys, this is something **really annoying** in the rocket.chat interface. It confuses people and gets in the way of the natural flow of conversations. It's been too many years. This should be prioritized over other less useful features, as it is a daily problem that our online communities face. Thanks.

**Comment by MrVonkey** [comment]
> C'mon guys, this is something **really annoying** in the rocket.chat interface. It confuses people and gets in the way of the natural flow of conversations. It's been too many years. This should be prioritized over other less useful features, as it is a daily problem that our online communities face. Thanks.

Yes absolutely, this should be prioritized. Unfortunately I couldn't find anything concerning this in the changelog for the 5.0.0 release candidate. Maybe I missed something, since there are several issues/requests related to this?

**Comment by MrVonkey** [comment]
Since version 5.0.0 there are more possibilities to customize the layout of the message component. You can try the following custom CSS in Settings > Layout to place the username / message header above the quoted message block and everything else below the quoted message block:

```css
.rcx-message-container > div {
 order: 2; 
}

.rcx-message-container > div.rcx-message-header {
  order: 0;
}

.rcx-message-container > div.rcx-attachment__content {
  order: 1;
}
```

**Comment by engelgabriel** [comment]
@ggazzo can we do that with the new message parser?

**Comment by ggazzo** [comment]
it's not a message parser issue, it's a design definition, which can be changed if it makes sense

**Comment by ankar84** [comment]
@engelgabriel @ggazzo it definitely make sense!! All that issue with many comments and reactions a good illustration of fact, community need that important improvement a lot!

**Comment by engelgabriel** [comment]
@ggazzo can you link the PR that fixed this here?

**Comment by ggazzo** [comment]
we will never have a pr who **fixes** this, because this is not a issue/bug. 

This is a design change, based on user input or feature/improvement request.

I will put the design team in the loop, if they believe this is the best option, I'm sure it will be implemented soon

btw this is something that @milton-rucks can help as well

**Comment by engelgabriel** [comment]
@dudanogueira did someone already work on this?

**Comment by milton-rucks** [comment]
I agree with the change. We are considering it for the next semester.

**Comment by dudanogueira** [comment]
> @dudanogueira did someone already work on this?

Oooops! For some reason I believed this was delivered in 5.4, as I mentioned at our monthly [Community Open Call ](https://open.rocket.chat/channel/Rocket-Chat-Open-Community)

But it was not. Sorry! But worry not! 

Now we have everything in place to deliver this and the dark mode ([that you can already preview with that cool Easter Egg](https://forums.rocket.chat/t/5-4-0-dark-theme/15504)) in 6.0

**Comment by rp-sg** [comment]
Is this possible yet?

**Comment by nguyennhatdung** [comment]
i think we shoud wait rocket chat version 100 for this...

**Comment by tmkis2** [comment]
I mentioned in on two community calls but nothing happened so far. :-(

**Comment by istankovic** [comment]
> A lot of users asked for it more then 2 years now.

Counting 5 years now...

**Comment by brlarini** [comment]
> > @dudanogueira did someone already work on this?

> Now we have everything in place to deliver this and the dark mode ([that you can already preview with that cool Easter Egg](https://forums.rocket.chat/t/5-4-0-dark-theme/15504)) in 6.0

We're already at 6.4.4 and this still isn't a thing...

**Comment by DRKMDMZ** [comment]
Well I guess it will be added sometime in the 6.4 or 6.5 update - but then it will bekome a "premium feature" like read receipts are.

**Comment by reetp** [comment]
Closing in favourt of

**Comment by casalsgh** [comment]
Since 2018 this ticket has been open. Thanks all for sticking with us to see it though - Took us a while, but we heard you and now it's closed, with new display of quoted messages going out on release 6.10.0. 

Thanks to our community liaison reetp (John), MartinSchoeler and folks from internal teams reviews for making it happen!

**Comment by linwalth** [comment]
Just realized this - THANK YOU! 👍
